Desktop/data/undostack.cpp

```diff
@@ -19,7 +19,12 @@ UndoStack::UndoStack(QObject* parent) : QUndoStack(parent)
 void UndoStack::pushCommand(UndoModelCommand *command)
 {
 	if (!_parentCommand) // Push to the stack only when no macro is started: in this case the command is autmatically added to the _parentCommand
-		push(command);
+	{
+		if (_busy)
+			Log::log() << "Try to push a new command (\"" << command->text() << "\") in the undostack while the undo or redo is not yet finished!!!" << std::endl;
```

Comment:
But arent we gonna miss errors now...

Shouldnt this be an `assert`

Reply:
that is not actually a reason, yopu are describing the cause of the crash correctly though:
- we were not checking whether useCustom was on already which added superfluous undostuff which crashes jasp
- same for emptyValues setting

I just fixed that.

After that we do not need `_busy` for anything at all.
In fact, it would just let us miss potential problems by avoiding the crash through "_busy".
And who knows what in the future might be done with this code?
Adding extra completely unecessary checks and booleans around that does not make any sense to me...